dataRetriever/blockchain/blockchain.go

```diff
@@ -51,6 +52,7 @@ func (bc *blockChain) SetGenesisHeader(genesisBlock data.HeaderHandler) error {
 }
 
 // SetCurrentBlockHeaderAndRootHash sets current block header pointer and the root hash
+// Question for review: perhaps also receive the blockHash as a parameter (since SetCurrentBlockHeaderAndRootHash and SetCurrentBlockHeaderHash are always called in pair, so that we set all of them in the same critical section)?
```

Comment:
:+1:

Reply:
can be done

Reply:
Even though they were always called in pair in production code, they weren't in some tests. In order to avoid breaking the tests, I did not apply this change at the moment. Is this all right?